server/events/pull_updater.go

```diff
@@ -41,3 +47,50 @@ func (c *PullUpdater) updatePull(ctx *command.Context, cmd PullCommand, res comm
 		ctx.Log.Err("unable to comment: %s", err)
 	}
 }
+
+type FeatureAwarePullUpdater struct {
+	PullUpdater
+	HidePrevPlanComments bool
+	VCSClient            vcs.Client
+	MarkdownRenderer     *MarkdownRenderer
+	GlobalCfg            valid.GlobalCfg
+	FeatureAllocator     feature.Allocator
+}
+
+func (f *FeatureAwarePullUpdater) updatePull(ctx *command.Context, cmd PullCommand, res command.Result) {
+	githubChecks, err := f.FeatureAllocator.ShouldAllocate(feature.GitHubChecks, ctx.HeadRepo.FullName)
+	if err != nil {
+		githubChecks = false
```

Comment:
why are we just swallowing the error here? How would we know if something was wrong? We should at least log it.